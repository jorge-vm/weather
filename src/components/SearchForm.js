import { useFormik } from "formik";
import * as Yup from "yup";

const SearchForm = ({ apiCall }) => {
  const formik = useFormik({
    initialValues: {
      city: "",
      units: "imperial",
    },
    validationSchema: Yup.object({
      city: Yup.string().required("City is Required"),
      units: Yup.string().required("Unit is Required"),
    }),
    onSubmit: (values) => {
      apiCall(values);
    },
  });
  return (
    <form onSubmit={formik.handleSubmit}>
      <div className="input-group">
        <span className="input-group-text">City</span>
        <input
          id="city"
          type="text"
          className="form-control"
          placeholder="Enter city name, i,e. Seattle"
          {...formik.getFieldProps("city")}
        />
        <select className="form-select" {...formik.getFieldProps("units")}>
          <option value="imperial">°F</option>
          <option value="metric">°C</option>
        </select>
      </div>
      {formik.touched.city && formik.errors.city ? (
        <div className="alert alert-warning m-2">{formik.errors.city}</div>
      ) : null}
      <br />
      <button type="submit" className="btn btn-primary">
        Submit
      </button>
    </form>
  );
};

export default SearchForm;
