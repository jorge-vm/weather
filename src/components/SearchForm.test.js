import { render, screen, waitFor } from "@testing-library/react";
import SearchForm from "./SearchForm";
import userEvent from "@testing-library/user-event";

test("renders SearchForm", () => {
  const { container } = render(<SearchForm />);
  expect(container.firstChild).toMatchSnapshot();
});

test("rendering and submitting a basic Formik form", async () => {
  const handleSubmit = jest.fn();
  render(<SearchForm apiCall={handleSubmit} />);

  userEvent.type(
    screen.getByPlaceholderText(/Enter city name, i,e. Seattle/i),
    "Miami"
  );

  userEvent.click(screen.getByRole("button", { name: /submit/i }));

  await waitFor(() => expect(handleSubmit).toHaveBeenCalled());
});
