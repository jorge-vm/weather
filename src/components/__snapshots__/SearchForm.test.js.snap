// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders SearchForm 1`] = `
<form>
  <div
    class="input-group"
  >
    <span
      class="input-group-text"
    >
      City
    </span>
    <input
      class="form-control"
      id="city"
      name="city"
      placeholder="Enter city name, i,e. Seattle"
      type="text"
      value=""
    />
    <select
      class="form-select"
      name="units"
    >
      <option
        value="imperial"
      >
        °F
      </option>
      <option
        value="metric"
      >
        °C
      </option>
    </select>
  </div>
  <br />
  <button
    class="btn btn-primary"
    type="submit"
  >
    Submit
  </button>
</form>
`;
